cmd/agent/subcommands/secret/command.go

```diff
@@ -32,55 +31,74 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 	cliParams := &cliParams{
 		GlobalParams: globalParams,
 	}
+
 	secretInfoCommand := &cobra.Command{
 		Use:   "secret",
-		Short: "Print information about decrypted secrets in configuration.",
+		Short: "Display information secrets in configuration.",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return fxutil.OneShot(showSecretInfo,
+				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
+				core.Bundle(),
+			)
+		},
+	}
+	secretRefreshCommand := &cobra.Command{
+		Use:   "refresh",
+		Short: "Refresh secrets in configuration.",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return fxutil.OneShot(secret,
-				fx.Supply(cliParams),
+			return fxutil.OneShot(secretRefresh,
 				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
 				core.Bundle(),
 			)
 		},
 	}
+	secretInfoCommand.AddCommand(secretRefreshCommand)
 
 	return []*cobra.Command{secretInfoCommand}
 }
 
-func secret(_ log.Component, config config.Component, _ *cliParams) error {
-	if err := util.SetAuthToken(); err != nil {
-		fmt.Println(err)
-		return nil
+func showSecretInfo(config config.Component) error {
+	r, err := callAPIEndpoint("/agent/secrets", config)
+	if err != nil {
+		return err
 	}
+	fmt.Println(string(r))
+	return nil
+}
 
-	if err := showSecretInfo(config); err != nil {
-		fmt.Println(err)
-		return nil
+func secretRefresh(config config.Component) error {
+	r, err := callAPIEndpoint("/agent/secret/refresh", config)
+	if err != nil {
+		return err
 	}
+	fmt.Printf("Secrets refresh: %s\n", r)
 	return nil
 }
 
-func showSecretInfo(config config.Component) error {
+func callAPIEndpoint(apiEndpointPath string, config config.Component) ([]byte, error) {
+	if err := util.SetAuthToken(); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
 	c := util.GetClient(false)
 	ipcAddress, err := pkgconfig.GetIPCAddress()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	apiConfigURL := fmt.Sprintf("https://%v:%v/agent/secrets", ipcAddress, config.GetInt("cmd_port"))
+	apiConfigURL := fmt.Sprintf("https://%v:%v%v", ipcAddress, config.GetInt("cmd_port"), apiEndpointPath)
```

Comment:
What about using the `url` package to build the URL?

```go
url := url.URL{
		Scheme:   "https",
		Host:     fmt.Sprintf("%v:%v", ipcAddress, config.GetInt("cmd_port")),
		Path:     "apiEndpointPath",
	}
```

Reply:
I like it, done.